posthog/temporal/data_imports/external_data_job.py

```diff
@@ -0,0 +1,219 @@
+import json
+import dataclasses
+import uuid
+import datetime as dt
+from typing import List
+from posthog.warehouse.data_load.pipeline import (
+    PIPELINE_TYPE_INPUTS_MAPPING,
+    PIPELINE_TYPE_RUN_MAPPING,
+    move_draft_to_production,
+    SourceSchema,
+)
+from posthog.warehouse.data_load.sync_table import is_schema_valid, SchemaValidationError
+
+from posthog.warehouse.models.external_data_job import ExternalDataJob
+from posthog.warehouse.external_data_source.jobs import (
+    create_external_data_job,
+    update_external_job_status,
+    get_external_data_source,
+)
+from posthog.warehouse.models.external_data_source import ExternalDataSource
+
+# TODO: remove dependency
+from posthog.temporal.batch_exports.base import PostHogWorkflow
+from posthog.temporal.common.heartbeat import AsyncHeartbeatDetails
+from temporalio import activity, workflow, exceptions
+from temporalio.common import RetryPolicy
+from asgiref.sync import sync_to_async
+
+
+@dataclasses.dataclass
+class CreateExternalDataJobInputs:
+    team_id: int
+    external_data_source_id: str
+
+
+@activity.defn
+async def create_external_data_job_model(inputs: CreateExternalDataJobInputs) -> str:
+    run = await sync_to_async(create_external_data_job)(  # type: ignore
+        team_id=inputs.team_id,
+        external_data_source_id=inputs.external_data_source_id,
+    )
+
+    return str(run.id)
+
+
+@dataclasses.dataclass
+class UpdateExternalDataJobStatusInputs:
+    id: str
+    run_id: str
+    status: str
+    latest_error: str | None
+
+
+@activity.defn
+async def update_external_data_job_model(inputs: UpdateExternalDataJobStatusInputs) -> None:
+    await sync_to_async(update_external_job_status)(  # type: ignore
+        run_id=uuid.UUID(inputs.id),
+        status=inputs.status,
+        latest_error=inputs.latest_error,
+    )
+
+
+@dataclasses.dataclass
+class ValidateSchemaInputs:
+    source_schemas: List[SourceSchema]
```

Comment:
nit: We don't need typing.List from Python 3.9 onwards.
```suggestion
    source_schemas: list[SourceSchema]
```